### ISSUE TITLE
The process '/bin/rm' failed with exit code 1

### ISSUE THREAD

**Issue by nomeata** [JamesIves/github-pages-deploy-action#241]
**Describe the bug**

The action claims it failed, even though it succesfully deployed the page.

**Reproduce**

Run the actions as configured in https://github.com/nomeata/kaleidogen/commit/4900875ff5f4d98eee7f741f7b9e0c7c8f755b65 I guess.

**Logs**

https://github.com/nomeata/kaleidogen/runs/567739736?check_suite_focus=true

```
Run JamesIves/github-pages-deploy-action@releases/v3
Checking configuration and starting deployment… 🚦
Deploying using Access Token… 🔑
Configuring git…
Git configured… 🔧
Starting to commit changes…
Changes committed to the gh-pages branch… 📦
Running post deployment cleanup jobs…
##[error]The process '/bin/rm' failed with exit code 1
Deployment Failed ❌
```

**Additional Comments**

Some files, maybe directories, that are in the `FOLDER` are read-only when this action runs. This could be causing this problem.

**Comment by nomeata** [comment]
This is likely the failing command:
https://github.com/JamesIves/github-pages-deploy-action/blob/2f7dc3b973d5d496a3059ad3b560ecc3e0133873/src/git.ts#L242

**Comment by nomeata** [comment]
So maybe this line can be instructed to not preserve all permissions?
https://github.com/JamesIves/github-pages-deploy-action/blob/2f7dc3b973d5d496a3059ad3b560ecc3e0133873/src/git.ts#L154

Or a blunt `chmod u+w -R ${temporaryDeploymentDirectory}` should do the job as well.

**Comment by JamesIves** [comment]
Interesting, I've not encountered this error before. Would you be willing to make a PR for this?

**Comment by JamesIves** [comment]
Released as part of `3.4.7`: https://github.com/JamesIves/github-pages-deploy-action/releases/tag/3.4.7
